File: skills/openclaw-trustroom/src/tools/present_credential.ts
// present_credential: produce a holder-bound Presentation blob for a target audience.

import type { JSONRpcClient } from "../client.js";

export interface PresentCredentialParams {
    credential_id: string;
    audience: { pubkey_hex: string };
}

export interface PresentCredentialResult {
    // base64-encoded Presentation bytes the caller hands to a remote room host.
    presentation_blob: string;
}

export const presentCredential = {
    name: "present_credential",
    description: "Build a holder-bound Verifiable Presentation tied to a specific audience and nonce. Returns the serialised presentation blob.",
    handler: async (
        params: PresentCredentialParams,
        ctx: { client: JSONRpcClient },
    ): Promise<PresentCredentialResult> => {
        // Forward to the Python sidecar via JSONRpcClient.call("present_credential", params).
        return ctx.client.call<PresentCredentialResult>("present_credential", params);
    },
};

File: integration/tools/present_credential.py
"""present_credential: produce a holder-bound Presentation blob for a target audience."""

from __future__ import annotations

from communication.channel.agent_channel import AgentChannel
from integration.rpc.methods import RpcMethod
from integration.rpc.schema import PresentCredentialParams, PresentCredentialResult


class PresentCredentialMethod(RpcMethod):
    """JSON-RPC method that builds a Presentation via CredentialPresenter."""

    name = "present_credential"
    params_schema = PresentCredentialParams
    result_schema = PresentCredentialResult

    async def __call__(
        self,
        channel: AgentChannel,
        params: PresentCredentialParams,
    ) -> PresentCredentialResult:
        # Resolve the CredentialPresenter from the channel, build Presentation, serialise to bytes.
        ...
